spring-native/src/main/java/io/opentelemetry/example/graal/CustomHints.java

```diff
@@ -0,0 +1,33 @@
+package io.opentelemetry.example.graal;
+
+import io.opentelemetry.exporter.otlp.logs.OtlpGrpcLogRecordExporterBuilder;
+import io.opentelemetry.exporter.otlp.metrics.OtlpGrpcMetricExporterBuilder;
+import io.opentelemetry.exporter.otlp.trace.OtlpGrpcSpanExporterBuilder;
+import org.springframework.aot.hint.MemberCategory;
+import org.springframework.aot.hint.RuntimeHints;
+import org.springframework.aot.hint.RuntimeHintsRegistrar;
+import org.springframework.aot.hint.TypeReference;
+
+public class CustomHints implements RuntimeHintsRegistrar {
+
+  @Override
+  public void registerHints(RuntimeHints hints, ClassLoader classLoader) {
+
+    hints
+        .reflection()
+        .registerType(
+            TypeReference.of(
+                "io.opentelemetry.instrumentation.spring.autoconfigure.aspects.InstrumentationWithSpanAspect"),
```

Comment:
@TDtianzhenjiu Is this configuration required to make `@WithSpan` works with a Spring Boot Native Image application? If it's the case, we could do the fix in the [opentelemetry-java-instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation) repository to implement it in the OpenTelemetry starter. If you are interested, I can tell you where to put the fix in the OpenTelemetry starter and where to write the non-regression test.